src/ucs/arch/riscv/atomic.h

```diff
@@ -0,0 +1,67 @@
+/**
+* Copyright (C) Tactical Computing Labs, LLC. 2022. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCS_RISCV64_ATOMIC_H_
+#define UCS_RISCV64_ATOMIC_H_
+
+#define UCS_DEFINE_ATOMIC_ADD(wordsize, suffix) \
+    static inline void ucs_atomic_add##wordsize(volatile uint##wordsize##_t *ptr, \
```

Comment:
do we really want to define these, or could rely on compiler __builtin_XX constructs?

Reply:
Did a little quick grep work and saw that these files have dependencies on the functions generated by the macros:

* src/ucp/wireup/wireup_ep.c
* src/ucs/memory/rcache.c
* src/ucs/async/async.c
* src/ucs/async/thread.c
* src/uct/sm/base/sm_ep.c

not sure what the right answer looks like; this implementation attempts to maintain consistency with the other architecture implementations.

Reply:
see https://github.com/openucx/ucx/blob/1a9873aa6475b434a43cf36978cd725c53c3cc61/src/ucs/arch/atomic.h#L17, maybe we can use generic/atomic.h for riscv as well